xarray/core/npcompat.py

```diff
@@ -371,3 +375,130 @@ def nancumprod(a, axis=None, dtype=None, out=None):
         """
         a, mask = _replace_nan(a, 1)
         return np.cumprod(a, axis=axis, dtype=dtype, out=out)
+
+
+    def normalize_axis_tuple(axis, ndim, argname=None, allow_duplicate=False):
+        """
+        Normalizes an axis argument into a tuple of non-negative integer axes.
+
+        This handles shorthands such as ``1`` and converts them to ``(1,)``,
+        as well as performing the handling of negative indices covered by
+        `normalize_axis_index`.
+
+        By default, this forbids axes from being specified multiple times.
+
+        Used internally by multi-axis-checking logic.
+
+        .. versionadded:: 1.13.0
+
+        Parameters
+        ----------
+        axis : int, iterable of int
+            The un-normalized index or indices of the axis.
+        ndim : int
+            The number of dimensions of the array that `axis` should be normalized
+            against.
+        argname : str, optional
+            A prefix to put before the error message, typically the name of the
+            argument.
+        allow_duplicate : bool, optional
+            If False, the default, disallow an axis from being specified twice.
+
+        Returns
+        -------
+        normalized_axes : tuple of int
+            The normalized axis index, such that `0 <= normalized_axis < ndim`
+
+        Raises
+        ------
+        AxisError
+            If any axis provided is out of range
+        ValueError
+            If an axis is repeated
+
+        See also
+        --------
+        normalize_axis_index : normalizing a single scalar axis
+        """
+        try:
+            axis = [operator.index(axis)]
+        except TypeError:
+            axis = tuple(axis)
+        axis = tuple(normalize_axis_index(ax, ndim, argname) for ax in axis)
```

Comment:
I looked inside numpy and found `normalize_axis_index` is written in C, but what it does is super simple.
I added our in-house version of `normalize_axis_index` and now looks working.